### PR TITLE
Remove files containing Macos meta-data

### LINKED ISSUE #3529: [question] Are there environment difference between the "developer workflow" and `conan create`?
While moving one of our _fat_ dependencies from prebuilt binaries (packaged through Conan ; ) to a recipe actually building the thing, we are encountering a complication we never encountered for our other recipes:

We wrote the recipe, following the recommendend [Package Development Flow](https://docs.conan.io/en/latest/developing_packages/package_dev_flow.html#package-development-flow). The end to end flow works without any error, notably `conan build` command.

Yet, when moving to the final step of calling `conan create` on the exact same recipe, the process fails during the `build()`  step.

Are there known difference between the environments used by the "developer workflow" and `conan create` command that we could investigate to find the source of this failure?

### PR DESCRIPTION
**Pull request by jgsogo**
Changelog: Feature: Remove files containing Macos meta-data (files beginning by `._`)

- [x] closes #3529 
- [ ] document `dot_clean`

### CODE REVIEW

**Comment by jgsogo** [comment]
In the test you can see how the offending file appears in the system. If a copy is performed it vanish. That is the difference between the two workflows: one is making a copy while the other one isn't.

Adnn explains everything about it:

> Finally, back to the initial topic of this issue:
> > why is there a difference in behaviour between between the "developer workflow" and conan create
>
> Because, to fully reproduce the expected recipe that would need to copy the source folder to modify bits of it, I added a manual cp step to the developer workflow, because I could not find any conan developer command to do this copy otherwise.
>
> And the cp shipped with OS X took care of handling those ._ (applying the attributes, and not copying them over to destination).
>
> So, here it is, a recipe that was working perfectly in "developer workflow" is breaking at build time with conan create.

File: conans/client/tools/files.py [hunk 1]
```diff
@@ -363,3 +363,14 @@ def unix2dos(filepath):
 
 def dos2unix(filepath):
     _replace_with_separator(filepath, "\n")
+
+
+def dot_clean(folder):
```
Comment (lasote) [inline conans/client/tools/files.py]: I would name it `dot_underscore_clean`

Comment (jgsogo) [reply conans/client/tools/files.py]: There is a built-in utility in Mac called `dot_clean`. I wanted to keep the same naming, and then you will ask me: "Why don't we use the command line?", and I will say 😒 ...

```
⇒  dot_clean -h    
usage: dot_clean  [-fhmnpsv] [--keep=[mostrecent|dotbar|native]] [directory ...]

 f : flat.	Disables recursive conversion.
 h : help.	Print this message.
 m : Always delete apple double files.
 n : Delete apple double file if there is no matching native file.
 p : Preserve apple double file
 s : Follow symbolic links.
 v : Verbose

 keep : Specify whether to keep the mostrecent, dotbar, or native.

```

Comment (lasote) [reply conans/client/tools/files.py]: Then call it "mac_dot_clean" :P

Comment (jgsogo) [reply conans/client/tools/files.py]: Shall I move it to `conans.client.tools.apple` and call it `apple_dot_clean`?

**Comment by jgsogo** [comment]
I won't, there is a checkbox left in the first comment 👍

**Pull request merged**